pyparsing/helpers.py

```diff
@@ -22,10 +22,13 @@ def delimitedList(expr, delim=",", combine=False):
         delimitedList(Word(hexnums), delim=':', combine=True).parseString("AA:BB:CC:DD:EE") # -> ['AA:BB:CC:DD:EE']
     """
     dlName = str(expr) + " [" + str(delim) + " " + str(expr) + "]..."
+
+    trailing_delim = Optional(delim) if allowTrailingDelim else Empty()
```

Comment:
Adding a trailing `Empty()` can do unintended whitespace skipping at the end of the expression when no trailing delimiter is used. Need to reimplement so that without `allowTrailingDelim=True`, you only get the expression as it currently is defined.  Probably refactor to something like:

    if not combine:
        delim = delim.suppress()
    
    delimited_list_expr = expr + (delim + expr)[...]
    if allowTrailingDelim:
        delimited_list_expr += Optional(delim)
    
    if combine:
        return Combine(delimited_list_expr)
    else:
        return delimited_list_expr